src/front/app/routes/changePassword.js

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import propTypes from 'prop-types';
+import { TextField, PrimaryButton, MessageBar, MessageBarType } from 'office-ui-fabric-react';
+import apiCall from '../apiCall';
+
+class ChangePasswordRoute extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+
+        this.state = {
+            email: '',
+            old_password: '',
+            new_password: '',
+            error_email: false,
+            error_old_password: false,
+            error_new_password: false,
+            error_passwords_equal: false,
+            success: false,
+        };
+    }
+    handleSubmit(event) {
+        event.preventDefault();
+        this.setState(prevState => ({
+            ...prevState,
+            error_email: false,
+            error_old_password: false,
+            error_new_password: false,
+            error_passwords_equal: false,
+            success: false,
+            isSubmitting: true,
+        }));
+        apiCall('post', 'auth/change-password', {
+            email: this.state.email,
+            oldPassword: this.state.oldPassword,
+            newPassword: this.state.newPassword,
+        })
+            .then(() => {
+                this.setState(prevState => ({
+                    ...prevState,
+                    success: true,
+                }));
+                setTimeout(() => {
+                    this.props.history.push('/login');
+                }, 2000);
+            })
+            .catch(err => {
+                if (err.response) {
+                    if (err.response.data === 'validation/email') {
```

Comment:
1. a bunch of ifs checking the same value can be replaced with a `switch`
2. maybe instead of checking the response and setting a flag, have an `error` state variable and here in the `.catch` you'll fill the error message directly?